### DISCUSSION TITLE
GPT prompts to focus answers on psycopg 3?

### DISCUSSION THREAD

**Discussion by GollyJer** [psycopg/psycopg#554] [Q&A]
Hey all. I'm curious if anyone has found good prompts for answers about psycopg 3.
It's hit or miss for me in BARD, and OpenAI ChatGPT 3.5. Bing Chat (GPT 4) is a little better, probably because it has the ability to search. I point them to the docs and say "I'm using the pscyopg 3.18 package" but they still very often answer with outdated items or mix approaches from 2 and 3.

Does anyone have some tips? Thanks! 

PS - my biggest tip is using ChatGPT 3.5 plus. Export your database schema and paste it into the prompt with "this is my database schema". All the answers SQL code work perfectly after. It often gives SQL code that is much more efficient than I would have created.

**Comment by dvarrazzo** [comment]
Considering that there's more than 15 years of examples about psycopg 2 and just a couple with psycopg 3, and the similarity between the two, I can see how things are not clear-cut right now for an automatic code generator that remixes SO answers.

OTOH, most of the work will be to compose an SQL query, so adapting from 2 to 3 shouldn't be that hard.
